components/brave_wallet_ui/constants/types.ts

```diff
@@ -30,15 +31,25 @@ const BraveKeyringsTypes = [
 ] as const
 export type BraveKeyrings = (typeof BraveKeyringsTypes)[number]
 
+export const FilecoinNetworkTypes = [
+  BraveWallet.FILECOIN_MAINNET, BraveWallet.FILECOIN_TESTNET
+] as const
+export type FilecoinNetwork = typeof FilecoinNetworkTypes[number]
+
 export type WalletAccountTypeName =
   | 'Primary'
   | 'Secondary'
   | 'Ledger'
   | 'Trezor'
 
-export interface WalletAccountType extends BraveWallet.AccountInfo {
-  id: string
+export interface AccountId extends Omit<BraveWallet.AccountId, 'keyringId'> {
   keyringId: BraveKeyrings
+}
+
+export interface WalletAccountType
+  extends Omit<BraveWallet.AccountInfo, 'accountId'> {
+  id: string
+  accountId: AccountId
   tokenBalanceRegistry: TokenBalanceRegistry
   nativeBalanceRegistry: TokenBalanceRegistry
```

Comment:
Redefining AccountId so it has typesafe keyringId field. Not sure if there is a more expressive way to do that. Would not be needed if we have keyringId as enum.